src/appleseed-max-impl/appleseedrenderer/renderersettings.cpp

```diff
@@ -149,6 +165,37 @@ const char* get_shader_override_type(const int shader_override_type)
     }
 }
 
+const char* get_lighting_engine_type(const int lighting_engine_type)
+{
+    switch (lighting_engine_type)
+    {
+      case 0:  return "pt";
+      case 1:  return "sppm";
+      default: return "pt";
```

Comment:
We must assert that the default case doesn't happen:

    default:
      assert(!"Invalid lighting engine type.");
      return "pt";

Same thing for the other similar functions.